src/server/frontend_xwayland/xwayland_wm_surface.cpp

```diff
@@ -127,11 +106,14 @@ void mf::XWaylandWMSurface::net_wm_state_client_message(uint32_t const (&data)[5
         TOGGLE = 2,
     };
 
-    ClientMessageData values{data};
-    auto const action = values.unpack<Action>();
-    auto const prop_a = values.unpack<xcb_atom_t>();
-    auto const prop_b = values.unpack<xcb_atom_t>();
-    auto const source_indication = values.unpack<SourceIndication>();
+    int i = 0;
+    auto const action = static_cast<Action>(data[i]);
+    i++;
+    auto const prop_a = static_cast<xcb_atom_t>(data[i]);
+    i++;
+    auto const prop_b = static_cast<xcb_atom_t>(data[i]);
+    i++;
+    auto const source_indication = static_cast<SourceIndication>(data[i]);
```

Comment:
Seems a bit verbose, but OK. 

----

Digressing, I'd prefer:
```c++
    auto const* pdata = data;
    auto const action = static_cast<Action>(*pdata++);
    auto const properties[] = { static_cast<xcb_atom_t>(*pdata++),  static_cast<xcb_atom_t>(*pdata++) };
    auto const source_indication = static_cast<SourceIndication>(*pdata++);
```
Using a `properties` array makes the following loop a little cleaner.